src/WebJobs.Script/ManagedDependencies/ManagedDependencyProvider.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.PowerShellWorker.ManagedDependency;
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script.ManagedDependencies
+{
+    internal class ManagedDependencyProvider : IManagedDependencyProvider
+    {
+        private const string ManagedDependenciesPropertiesName = "managedDependencies";
+
+        /// <summary>
+        /// Returns the managed dependecies meta data
+        /// </summary>
+        /// <param name="hostFilePath">The configuration path for the managed dependencies</param>
+        /// <param name="oSPlatform">The oSPlatform the code is running on</param>
+        /// <param name="managedDependencyConfigs">Configurations for managed dependencies</param>
+        /// <returns>The IEnumerable list of <see cref="Dependency"/></returns>
+        public async Task<IEnumerable<Dependency>> GetDependencies(string hostFilePath,
+            OSPlatform oSPlatform,
+            IManagedDependencyConfigs managedDependencyConfigs)
+        {
+            if (string.IsNullOrWhiteSpace(hostFilePath)
+                || !File.Exists(hostFilePath))
+            {
+                return null;
+            }
+
+            var managedDependenciesFromHostFile = await DesirializeJsonFile<DependentLibraries>(hostFilePath);
```

Comment:
nit: change from `var` to the type name here (since it's the result of a method call and not evident)
nit: Spelling (`Deserialize`)